schemas/documentSchemas/website.ts

```diff
@@ -31,7 +31,20 @@ export default defineType({
       type: "boolean",
       initialValue: false,
     }),
-    urlField,
+    defineField({
+      ...urlField,
+      validation: (Rule) =>
+        Rule.custom((resourceUrl) => {
+          const pattern = new RegExp("http[s]*://[^/]+(/.+)");
+          if (typeof resourceUrl === "undefined") {
+            return true;
+          } else if (pattern.test(resourceUrl)) {
+            return "Warning: It is more likely that this URL points to a resource rather than a website.";
+          } else {
+            return true;
+          }
```

Comment:
You're good to merge this but a quick tip. Focus on readability and minimise nesting. JavaScript makes this really easy. In the future consider something like this:
```js
// We can keep this separate because we're explicitly dealing with the value being undefined
if (typeof resourceUrl === "undefined") {
  return true;
}

// By breaking out the next bit of work it's clear to future devs we're solving two separate problems
return pattern.test(resourceUrl) ? "Warning..." : true;
```

Reply:
Yeah good tip. I don’t think I fully understand what the undefined check is doing, where we return true, not sure what that does. That’s why I thought it should be linked to the rest of the conditional checks using the else/if statement.

Reply:
That's fair. You won't see a lot of checking for 'undefined' in tutorials and the like - because the teacher is normally confident what type the value will be. In real life, we can rarely expect a value to be of the type we expect.

In this case it's all to do with the fact that the `Rule.custom(value)` function is going to get called every time, whether or not the field has a value. If the field has no value then what we'll be getting is `undefined` instead of a `resourceUrl`. `pattern.test` coerces any type of value to a string, so we'd end up testing against the string "undefined" which is definitely not what we want.

So in plain terms what we're saying is - before we do any work - let's make sure that the value is of the type we expect.